src/sentry/services/hybrid_cloud/notifications/impl.py

```diff
@@ -142,6 +152,38 @@ def get_settings_for_user_by_projects(
             )
         ]
 
+    def get_setting_options_by_project(
+        self,
+        *,
+        user_id: int,
+        projects: Iterable[Project],
+        type: NotificationSettingEnum,
+    ) -> MutableMapping[
+        NotificationScopeEnum,
+        MutableMapping[int, MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum]],
+    ]:
+        """
+        Returns a a mapping of project scopes settings, with projects IDs
+        mapped to a map of provider to notifications setting values.
+        """
+        user = User.objects.get(id=user_id)
+        project_to_settings: MutableMapping[
+            int, MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum]
+        ] = defaultdict(dict)
+
+        # TODO(snigdha): This is an N+1 query. We should optimize this if there's a visible performance impact.
+        for project in projects:
+            providers = get_setting_providers_for_users(
+                [user_id],
+                project=project,
+                organization=project.organization,
```

Comment:
@snigdhas isn't this going to limit the query to the org scope? I'd think we just want to pass in the project here

Reply:
Hmm even if we don't pass the org, we derive the org [here](https://github.com/getsentry/sentry/blob/fd14efac64442442bebc99bab0cb3589be34b37c/src/sentry/notifications/helpers.py#L704). Are you saying we should skip that and only look at project scoped settings?

Reply:
I guess we OR the queries so it's fine!